### ISSUE TITLE
Invisible widgets take space

### ISSUE THREAD

**Issue by amin-nas** [EnsembleUI/ensemble#181]
Widgets that are invisible take space instead of being completely invisible.

**Comment by kmahmood74** [comment]
please provide a test case in the form of a EDL definition so this may be assigned to someone who can just get it from the description.

**Comment by amin-nas** [comment]
Test case:
```yaml
View:
  Column:
    styles: { gap: 8, backgroundColor: yellow }
    children:
      - Column:
          visible: false
          children:
            - Text:
                text: I am invisible
      - Column:
          visible: false
          children:
            - Text:
                text: I am invisible
      - Column:
          visible: false
          children:
            - Text:
                text: I am invisible
      - Column:
          visible: false
          children:
            - Text:
                text: I am invisible
      - Column:
          visible: false
          children:
            - Text:
                text: I am invisible
      - Column:
          visible: true
          children:
            - Text:
                text: I am visible
```
<img width="969" alt="Screenshot 2023-01-25 at 11 30 33 AM" src="https://user-images.githubusercontent.com/491202/214505375-7663dd27-c2ac-4d80-ba1a-f16510810f67.png">

**Comment by vinothzyngl** [comment]
It took space because of adding a `gap: 8` in the styles of column @amin-nas

**Comment by vinothvino42** [comment]
This is fixed in this PR #370
